openlibrary/core/models.py

```diff
@@ -495,6 +495,16 @@ def get_users_notes(self, username, edition_olid=None):
         edition_id = extract_numeric_id_from_olid(edition_olid) if edition_olid else -1
         return Booknotes.get_patron_booknote(username, work_id, edition_id=edition_id)
 
+    def has_book_note(self, username, edition_olid):
```

Comment:
performance implications of hitting books to get whether note exists? Seems like it's fine if we may show the text on the page somewhere anyway.